crates/wysiwyg/src/tests/test_lists.rs

```diff
@@ -112,11 +112,34 @@ fn entering_mid_text_node_with_selection() {
 #[test]
 fn removing_list() {
     let mut model = cm("|");
-    model.create_ordered_list();
+    model.ordered_list();
     model.enter();
     assert_eq!(tx(&model), "|");
 }
 
+#[test]
+fn updating_list_type() {
+    let mut model = cm("<ol><li>ab</li><li>cd|</li></ol>");
+    model.unordered_list();
+    assert_eq!(tx(&model), "<ul><li>ab</li><li>cd|</li></ul>");
+    model.ordered_list();
+    assert_eq!(tx(&model), "<ol><li>ab</li><li>cd|</li></ol>");
+}
+
+#[test]
+fn moving_list_item_content_out() {
+    let mut model = cm("<ol><li>ab</li><li>cd|</li></ol>");
+    model.ordered_list();
+    assert_eq!(tx(&model), "<ol><li>ab</li></ol>cd|");
+}
+
+#[test]
+fn appending_new_list_to_previous() {
+    let mut model = cm("<ol><li>ab</li></ol>cd|");
+    model.ordered_list();
+    assert_eq!(tx(&model), "<ol><li>ab</li><li>cd|</li></ol>");
+}
+
```

Comment:
This all looks great, but there is one testcase that I think it currently does wrong. Here is my attempt at expressing it in Rust:

```rust
let mut model = cm("<ul><li>foo|</li></ul>bar");
model.ordered_list();
assert_eq!(tx(&model), "<ul><li>foo</li></ul><ol><li>​bar|</li></ol>");
```

Currently it ends up like:

```rust
assert_eq!(tx(&model), "<ol><li>foo</li><li>​bar|</li></ol>");
```

I.e. it changes the preceding list into `ol`. I think it should not do that, but instead make a new `ul`.

Reply:
(I just confirmed in Google Docs it works the way I think we should work)

Reply:
I added this test case (slightly different as we don't insert ZWSP right now - better do it on a separated piece of work since it might be tricky and mess up selection)